packages/react/src/primitives/shared/utils.ts

```diff
@@ -193,3 +196,27 @@ export const classNameModifierByFlag = (
 ): string => {
   return flag ? `${base}--${modifier}` : null;
 };
+
+export const getCSSVariableIfValueIsThemeKey = (
+  propKey: string,
+  value: string | number
+) => {
+  if (typeof value === 'number') {
+    return value;
+  }
+  const path = value.split('.');
+  let { tokens } = defaultTheme;
+  tokens = tokens[stylePropsToThemeKeys[propKey]];
+  for (let i = 0; i < path.length; i++) {
+    if (tokens) {
+      tokens = tokens[path[i]];
```

Comment:
I'm having trouble understanding what exactly this line is doing. Would you mind explaining?

Reply:
This will go deep into the theme `tokens` and get its `value` and also works as a check to see if the key is valid because users can provide any string key. If a key is invalid, then you will get nothing but `undefined` from the `tokens` we defined